srtcore/socketconfig.h

```diff
@@ -0,0 +1,1140 @@
+    int m_iRcvBufSize;                           // Maximum UDT receiver buffer size
+    linger m_Linger;                             // Linger information on close
+    bool m_bRendezvous;                          // Rendezvous connection mode
+
+    duration m_tdConnTimeOut;    // connect timeout in milliseconds
+    bool m_bDriftTracer;
+    int m_iSndTimeOut;                           // sending timeout in milliseconds
+    int m_iRcvTimeOut;                           // receiving timeout in milliseconds
+    int64_t m_llMaxBW;                           // maximum data transfer rate (threshold)
+
+    // These fields keep the options for encryption
+    // (SRTO_PASSPHRASE, SRTO_PBKEYLEN). Crypto object is
+    // created later and takes values from these.
+    HaiCrypt_Secret m_CryptoSecret;
+    int m_iSndCryptoKeyLen;
+
+    // XXX Consider removing. The m_bDataSender stays here
+    // in order to maintain the HS side selection in HSv4.
+    bool m_bDataSender;
+
+    bool m_bMessageAPI;
+    bool m_bTSBPD;               // Whether AGENT will do TSBPD Rx (whether peer does, is not agent's problem)
+    int m_iRcvLatency;           // Agent's Rx latency
+    int m_iPeerLatency;       // Peer's Rx latency for the traffic made by Agent's Tx.
+    bool m_bTLPktDrop;           // Whether Agent WILL DO TLPKTDROP on Rx.
+    int m_iSndDropDelay;         // Extra delay when deciding to snd-drop for TLPKTDROP, -1 to off
+    bool m_bEnforcedEnc;    // Off by default. When on, any connection other than nopw-nopw & pw1-pw1 is rejected.
+    int m_GroupConnect;
+    int m_iPeerIdleTimeout;      // Timeout for hearing anything from the peer.
+    uint32_t m_uStabilityTimeout;
+    int m_iRetransmitAlgo;
+
+    int64_t m_llInputBW;                         // Input stream rate (bytes/sec)
+    // 0: use internally estimated input bandwidth
+    int m_iOverheadBW;                           // Percent above input stream rate (applies if m_llMaxBW == 0)
+    bool m_bRcvNakReport;                        // Enable Receiver Periodic NAK Reports
+    int m_iMaxReorderTolerance;                  //< Maximum allowed value for dynamic reorder tolerance
+
+    // For the use of CCryptoControl
+    // HaiCrypt configuration
+    unsigned int m_uKmRefreshRatePkt;
+    unsigned int m_uKmPreAnnouncePkt;
+
+    uint32_t m_lSrtVersion;
+    uint32_t m_lMinimumPeerSrtVersion;
+
+    StringStorage<MAX_CONG_LENGTH>    m_Congestion;
+    StringStorage<MAX_PFILTER_LENGTH> m_PacketFilterConfig;
+    StringStorage<MAX_SID_LENGTH>     m_StreamName;
+
+    // Shortcuts and utilities
+    int32_t flightCapacity()
+    {
+        return std::min(m_iRcvBufSize, m_iFlightFlagSize);
+    }
+
+    CSrtConfig()
+        : m_iMSS(DEF_MSS)
+          , m_zExpPayloadSize(SRT_LIVE_DEF_PLSIZE)
+          , m_bSynSending(true)
+          , m_bSynRecving(true)
+          , m_iFlightFlagSize(DEF_FLIGHT_SIZE)
+          , m_iSndBufSize(DEF_BUFFER_SIZE)
```

Comment:
Alignment
```suggestion
    CSrtConfig()
        : m_iMSS(DEF_MSS)
        , m_zExpPayloadSize(SRT_LIVE_DEF_PLSIZE)
        , m_bSynSending(true)
        , m_bSynRecving(true)
        , m_iFlightFlagSize(DEF_FLIGHT_SIZE)
        , m_iSndBufSize(DEF_BUFFER_SIZE)
```
and so on.